src/vault/serializers/vault-secret.serializer.ts

```diff
@@ -0,0 +1,81 @@
+import { List, ListResponse } from '../../common/interfaces';
+import {
+  ReadSecretMetadataResponse,
+  ReadSecretResponse,
+  UpdateSecretOptions,
+  UpdateSecretEntity,
+  SecretMetadata,
+  VaultSecret,
+  SecretVersionResponse,
+  SecretVersion,
+  CreateSecretOptions,
+  CreateSecretEntity,
+  SecretDigestResponse,
+  SecretDigest,
+  ListSecretVersionsResponse,
+} from '../interfaces';
+
+export const deserializeSecretMetadata = (
+  metadata: ReadSecretMetadataResponse,
+): SecretMetadata => ({
+  context: metadata.context,
+  environmentId: metadata.environment_id,
+  id: metadata.id,
+  keyId: metadata.key_id,
+  updatedAt: new Date(Date.parse(metadata.updated_at)),
```

Comment:
If `metadata.updated_at` is ISO-8601, this will do the trick:
```suggestion
  updatedAt: new Date(metadata.updated_at),
```

Reply:
How well does that handle varying time resolution? In some instances the Vault API return millisecond resolution, in others microsecond.

Reply:
Good question. Microseconds...poorly. Though as far as I'm aware it would be the same with Date.parse. Pure JS is bad with microseconds in general as far as I understand, but did you find success with `new Date(Date.parse())`?

Reply:
Well it's more about being defensive. `Date.parse` will return `NaN` for formats it can't understand, while `new Date` will return a Date a epoch for formats it can't understand. I'd prefer the value to be null (likely to raise an error) than epoch (might go unnoticed). But I'm open to other approaches

Reply:
It might only be for cases of `null` - 
```
const funny = null
new Date(Date.parse(funny)) // returns null
new Date(funny) // returns epoch
```